### DISCUSSION TITLE
How do I check for no hard links for all categories?

### DISCUSSION THREAD

**Discussion by vincejv** [StuffAnThings/qbit_manage#456] [Q&A]
How do I check hard links for all categories?
My config file is like this
```yaml
nohardlinks:
  # Tag Movies/Series that are not hard linked outside the root directory
  # Mandatory to fill out directory parameter above to use this function (root_dir/remote_dir)
  # This variable should be set to your category name of your completed movies/completed series in qbit. Acceptable variable can be any category you would like to tag if there are no hardlinks found
- All
- Uncategorized
```
but it doesn't work, any work arounds?

**Comment by bakerboy448** [comment]
The [wiki](https://github.com/StuffAnThings/qbit_manage/wiki/Config-Setup) and config file example list everything that's required. Simply adding the hardlinks section doesn't do anything nor is valid.

Are these categories configured in QBm and the name of them in qbit?

**Reply by vincejv** [reply]
sorry i revised the post, what I meant is, how do I check for "nohardlinks" for All Categories, i tried to use `All` and `Uncategorized` but it doesn't detect any torrents, justs says 0 torrents.

**Reply by vincejv** [reply]
but when i specify the "actual" category, it works

**Reply by bakerboy448** [reply]
Correct because you need to specify all categories in QBm and the specific not aggregate categories for nohl

**Reply by vincejv** [reply]
okay... how do i include the torrents with no category?

**Reply by bakerboy448** [reply]
There's no `All` aggregate value — `nohardlinks:` iterates your list and calls qBittorrent's API with each entry as an exact `category` filter (`modules/core/tag_nohardlinks.py`), so `All` and `Uncategorized` (as literal strings) match zero torrents because no qBittorrent category is actually named that.

For uncategorized torrents specifically, use an empty string as the key — that's the real qBittorrent Web API sentinel for "no category" (same mechanism qbm's own category-update feature uses internally to query uncategorized torrents — `modules/core/category.py`):
```yaml
nohardlinks:
  movies:
  tv:
  "":   # torrents with no category set
```
For "every category," there's no wildcard — list each category explicitly (`movies`, `tv`, `""`, etc.).
